db/description/collection.go

```diff
@@ -0,0 +1,214 @@
+// Copyright 2023 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package description
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/ipfs/go-datastore/query"
+
+	"github.com/sourcenetwork/defradb/client"
+	"github.com/sourcenetwork/defradb/core"
+	"github.com/sourcenetwork/defradb/datastore"
+)
+
+func SaveCollection(
+	ctx context.Context,
+	txn datastore.Txn,
+	desc client.CollectionDescription,
+) (client.CollectionDescription, error) {
+	buf, err := json.Marshal(desc)
```

Comment:
A go standard is also just to use `b` in a case like this.

Reply:
I would argue that short var names like this are only acceptable for very short functions (I'd say up to 6 lines of code) otherwise it doesn't do any good to readability.

Reply:
To me as long as you can see the whole function on a smallish screen. then it's fine. That probably means around 50 lines. But definitely more than 6. As long as there aren't many more byte arrays defined. That being said, it was only a option. Your suggestions are good as well.

Reply:
We use `buf` all over the place in our codebase, and I'm pretty sure I've seen it for stuff like this in a fair few other codebases.  Islam, internally within `json.Marshal` the return value probably does fit your `some kind of preallocated memory` description and that's how I've seen this naming come from.

Let me know if either of you think this is important, as I don't, and without clear consensus between you guys I'll leave it as is otherwise.

Reply:
`buf` is misrepresenting what this is but we do have a few places where we use it probably by mistake including one that Islam wrote
<img width="549" alt="Screenshot 2023-10-24 at 2 42 21 PM" src="https://github.com/sourcenetwork/defradb/assets/4949751/b523c3f8-a879-4e1d-b9f0-a257c60c608b">
but most were added by Andy in  #1965. `buf` is usually used for `bytes.Buffer` as we do a lot in the code base. 

My opinion is that it's the wrong name for the variable but I wouldn't say its important to me in this case. It was good to point it out though.

Reply:
> but most were added by Andy in https://github.com/sourcenetwork/defradb/pull/1965.

nitpick: If you look at `db/collection.go` in that PR, you should see that these variables were `fooBuf` before :)

Reply:
Fair enough 🙂